database2/cassandraKS.js

```diff
@@ -0,0 +1,39 @@
+const cass = require('cassandra-driver');
+
+const client = new cass.Client({
+  contactPoints: ['localhost'],
+  localDataCenter: 'datacenter1',
+  keyspace: 'system'
+});
+
+const createProductsKeyspace = () => {
+  client.connect()
+    .then(() => {
+      const query = "DROP KEYSPACE similar_prod";
+      return client.execute(query);
+    })
+    .catch((err) => {
+      console.error('keyspace not defined or other error: ', err);
+    })
+    .then((result) => {
+      console.log('keyspace dropped: ', result);
+      const query = "CREATE KEYSPACE IF NOT EXISTS similar_prod WITH replication =" + "{'class': 'SimpleStrategy', 'replication_factor': '1'}";
+      return client.execute(query);
+    })
+    .then((result) => {
+      console.log('keyspace created: ', result);
+      const query = "CREATE TABLE IF NOT EXISTS similar_prod.products" + "(productId int PRIMARY KEY, name text, category text, price decimal, rating decimal, imageUrl text, onSale boolean)"
```

Comment:
A note on your schema here: A CSV file is basically just a .txt file. In other words, all of your information will be in the form of text/strings. I think that you would have to do conversion to set your table up like this. However, I've been looking into Cassandra's ability to format a query result as JSON. So there's the possibility of using JSON.parse or something like it to 'revive' your data on the server before you send it back to the client. Just a thought!

Reply:
Check this out if you're interested!
https://cassandra.apache.org/doc/latest/cql/json.html

Reply:
good point! I don't think I have much understanding of what Cassandra does to incoming data types. Definitely have some questions about when to REVIVE! Thanks for the feedback!